pkg/controller/machine/controller.go

```diff
@@ -82,6 +109,27 @@ func newReconciler(mgr manager.Manager, actuator Actuator) reconcile.Reconciler
 	return r
 }
 
+func stringPointerDeref(stringPointer *string) string {
+	if stringPointer != nil {
+		return *stringPointer
+	}
+	return ""
+}
+
+func predicateUpdateFunc(e event.UpdateEvent) bool {
+	oldObject := e.ObjectOld.(*machinev1.Machine)
```

Comment:
I guess it's fine here, but of course this will panic if this type assertion fails.